### PR TITLE
Enable doc building on CircleCI

### PR DESCRIPTION
**Pull request by QuLogic**
Enable doc building on CircleCI using a method similar to Matplotlib. It builds on both Python 2 and 3. Warnings are not fatal at the moment, but could be made so (probably after they're fixed though.) The last step in the build prints out a link that you can use to view the rendered docs. Here's the link for [Python 2](https://circleci.com/gh/QuLogic/cartopy/18/artifacts/0//root/project/docs/build/html/index.html) and for [Python 3](https://circleci.com/gh/QuLogic/cartopy/19/artifacts/0//root/project/docs/build/html/index.html). Someone with admin privileges will have to enable CircleCI for this project before this is merged.

I already see a few bugs:
* The [Interrupted Goode Homolosine plot](https://19-21996491-gh.circle-artifacts.com/0/root/project/docs/build/html/crs/projections.html#interruptedgoodehomolosine) seems to have only one gridline.
* There are some extra `REMOVE` pages in the gallery. I'm not sure if @choldgraf figured out how to get rid of the `__init__.py` files when rendered. Maybe we can get rid of them now that we have CircleCI testing examples instead.
* The [circular boundary example](https://19-21996491-gh.circle-artifacts.com/0/root/project/docs/build/html/gallery/lines_and_polygons/always_circular_stereo.html) is missing the features (both 2 and 3).
* ~~"The effect of badly referencing an ellipse" works on [Python 2](https://18-21996491-gh.circle-artifacts.com/0/root/project/docs/build/html/gallery/lines_and_polygons/effects_of_the_ellipse.html), but is missing the features on [Python 3](https://19-21996491-gh.circle-artifacts.com/0/root/project/docs/build/html/gallery/lines_and_polygons/effects_of_the_ellipse.html).~~ FIXED

### CODE REVIEW

**Comment by pp-mo** [comment]
Totally awesome, this is just what we want 🥇 !

**Comment by pp-mo** [comment]
Question : why bother building docs in both Python2 and Python3 ?
(and testing docs, and maintaining them)
1) we already test cartopy *itself* with both
2) we can reasonably standardise the docs-build process for Python 3 only ?

**Comment by pp-mo** [comment]
Update info: 
 * the circular-boundary problem, at least, is pre-existing.  It already occurs with matplotlib 1.5 and python 2.7.  *So fixing that is not in scope for this.*
 * other problems do seem to relate to Python3 specifically, and not matplotlib 2 :
   * the missing-features problems 
   * the vanished meridians on the InterruptedGoodeHomolosine example
   * various gridline behaviours are also different : the default spacings are off and the linestyle doesn't default to dotted

**Comment by QuLogic** [comment]
Half the Travis builds are pinned, the others are not, so I'm not sure it's necessary. Besides, it makes things use the old theme and won't look anything like it does for people who install it now. You're already seeing the differences and getting tricked into thinking they're bugs in the current version. The gridlines are supposed to be solid and the spacing from before was wrong.

I think it'd be fine to standardize on Python 3 only; I don't think it's useful to do the reverse.

Almost 50% of the build time is taken up by the `geostationary` example. Probably some good incentive to speed up the reprojection. We could also attempt to cache some of the feature datasets, though the conditions for that are a bit more rigid than Travis.

I'm fine with merging this as-is and fixing the remaining things in separate PRs.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+doc-run: &doc-build
+  name: Build documentation
+  command: |
+    source activate test-environment
+    make html
+  working_directory: docs
+
+doc-bundle-run: &doc-bundle
+  name: Bundle sphinx-gallery documentation artifacts
```
Comment (pelson) [inline .circleci/config.yml]: Any good reason for doing this?

**Comment by pelson** [comment]
Ultimately I'd want to follow a similar pattern to Iris and push to scitools.org.uk whenever we have a branch or tag, but this is 95% of the way to achieving that 👍

**Comment by pelson** [comment]
The license header years need updating too.

**Comment by pelson** [comment]
I've enabled circle on scitools/cartopy. I'm very tempted to merge this branch as is (I can update the copyright year directly if it is a problem introduced here, which I don't think it is)

**Comment by pelson** [comment]
> The license header years need updating too.

Think that is a by-product of something else. I'll ensure it is fixed (if you haven't already!). 😄

**Comment by pelson** [comment]
Wowzers. What an awesome piece of work. Thanks @QuLogic! 

Hopefully this will help us all to see where we are at, and what needs doing before we can cut the next release (soon I expect).

Thanks a bunch for doing this! 👍 🍻

File: .circleci/config.yml (hunk 1 above)
Comment (QuLogic) [reply .circleci/config.yml]: Mostly to see what's wrong with the examples if the result comes out weird.

**Pull request merged**